js/src/setup-mc/setup-stepper/setup-free-listings/index.js

```diff
@@ -59,8 +70,40 @@ const SetupFreeListings = ( props ) => {
 		return {};
 	};
 
-	const handleSubmitCallback = () => {
-		onContinue();
+	const handleSubmitCallback = async ( values ) => {
+		/**
+		 * Even though we already have auto-save effects in the FormContent,
+		 * we are saving the form values here again to be sure,
+		 * because the auto-save may not be fired
+		 * when users were having focus on the text input fields
+		 * and then immediately click on the Continue button.
+		 */
+		const {
+			shipping_country_rates: shippingRatesValue,
+			...settingsValue
+		} = values;
+
+		setSaving( true );
+
+		try {
+			await Promise.all( [
+				saveSettings( settingsValue ),
+				saveShippingRates( shippingRatesValue ),
+			] );
+			onContinue();
+		} catch ( error ) {
+			createNotice(
+				'error',
+				__(
+					'There is a problem in saving your settings and shipping rates. Please try again later.',
+					'google-listings-and-ads'
+				)
+			);
+		}
```

Comment:
💅 
It's almost impossible to enter this catch block because `saveSettings` and `saveShippingRates` already catch all errors inside.

Reply:
> `saveSettings` and `saveShippingRates` already catch all errors inside.

Ahh right. I'm leaning towards keeping this intact, and feeling like one day we should remove the `handleFetchError` in `actions.js`, or at least let the error propagate up to components so that the components are able to handle the error in the component if they want to. 📜 WDYT?

Reply:
Thanks for the explanation!

> WDYT?

I prefer to see how to handle error catching in the future when the error handling in *.~/data* is adjusted, otherwise when reading the code devs might wonder why there is a catch that almost never works, or spend time thinking whether there is an edge case. Or, at least add a code comment for the reason why the `catch` is here but does not work.

Reply:
> I prefer to see how to handle error catching in the future when the error handling in .~/data is adjusted

I gave a few more thoughts on this and I thought of an idea: I can look into removing the try/catch block for `saveShippingRates` (refactor it with `deleteShippingRates` and `upsertShippingRates`) in actions.js in https://github.com/woocommerce/google-listings-and-ads/issues/1287, so that when there is an error in those shipping rates operations, the error would bubble up and can be handled by the components. The error will be caught by the catch block we are discussing here. Does that sound good? 🙂

Reply:
Sound great! Thanks!